processor/metricstransformprocessor/metrics_transform_processor_otlp.go

```diff
@@ -267,22 +265,37 @@ func (mtp *metricsTransformProcessor) processOTLPMetrics(md pmetric.Metrics) (pm
 					}
 
 					extractedMetrics := extractAndRemoveMatchedMetrics(transform.MetricIncludeFilter, metrics)
-					combinedMetric := metrics.AppendEmpty()
-					combine(transform, extractedMetrics).MoveTo(combinedMetric)
-
-					// set matchedMetrics to the combined metric so that any additional operations are performed on
-					// the combined metric
-					matchedMetrics = []pmetric.Metric{combinedMetric}
-				}
-
-				for _, metric := range matchedMetrics {
-					if transform.Action == Insert {
-						matchedMetric := transform.MetricIncludeFilter.extractMatchedMetric(metric)
-						newMetric := metrics.AppendEmpty()
-						matchedMetric.CopyTo(newMetric)
-						metric = newMetric
+					combinedMetric := combine(transform, extractedMetrics)
+					if transformMetric(combinedMetric, transform) {
+						combinedMetric.MoveTo(metrics.AppendEmpty())
+					}
+				case Insert:
+					newMetrics := pmetric.NewMetricSlice()
+					newMetrics.EnsureCapacity(metrics.Len())
+					for i := 0; i < metrics.Len(); i++ {
+						metric := metrics.At(i)
+						newMetric := transform.MetricIncludeFilter.extractMatchedMetric(metric)
+						if newMetric == (pmetric.Metric{}) {
+							continue
+						}
+						if newMetric == metric {
+							newMetric = pmetric.NewMetric()
+							metric.CopyTo(newMetric)
+						}
+						if transformMetric(newMetric, transform) {
+							newMetric.MoveTo(newMetrics.AppendEmpty())
+						}
 					}
-					transformMetric(metric, transform)
+					newMetrics.MoveAndAppendTo(metrics)
+				case Update:
```

Comment:
Is Update a new case?  I don't see it in the original

Reply:
It was merged with `Insert` case before

Reply:
Do we need a new default case?  Maybe something that returns an error?

Reply:
no, each case has isolated logic now and we cover all of them